tutorials/1_basic_square/README.md

```diff
@@ -0,0 +1,153 @@
+  // will work fine for our small example and will make more sense when we implement
+  // the update method
+  int squareDirection = 1;
+
+  // The onLoad method is where all of the game initialization is supposed to go
+  // for this example, you may think that this square could just be initialized on the field
+  // declaration, and you are right, but for learning purposes and to present the life cycle method
+  // for this example we will be initializing this field here.
+  @override
+  Future<void> onLoad() async {
+    squarePos = Rect.fromLTWH(0, 0, 100, 100);
+  }
+
+  // Update and render ommited
+}
+```
+
+Right, now we have all the data and variables we need to start implementing our game, for the next step, lets just draw our little square on the screen.
+
+```dart
+  @override
+  void render(Canvas canvas) {
+    // Canvas is a class from dart:ui and is reponsible for all the renderings on Flame
+    //
+    // BasicPalette is a help class from Flame, which provides default, pre built instances
+    // of Paints that can be used by your game
+    canvas.drawRect(squarePos, BasicPalette.white.paint);
+  }
+}
+```
+
+You may now be seeing a static white square being rendered on the top left corner of your screen, which is not that cool, so, to finish our example, lets add some movement to the game and implement our update method:
+
+```dart
+  @override
+  void update(double dt) {
+    // Here we move our square by applying our iteration delta time (dt) to our
+    // speed variable and direction. Note that the Rect class class is immutable
+    // and the translate method returns a new Rect instance for us, so we just
+    // re assign it to our square variable
+    //
+    // It is important to remember that the result of the execution of this method,
+    // must be the game state (in our case our rect, direction variabables) updated to be
+    // consistent of how it should be after the amount of time stored on the dt variable,
+    // that way your game will always run smooth and consistent even when a FPS drop or peak happen.
+    //
+    // To illustrate this, if our square moves at 200 logical pixels per second, and half a second
+    // has passed, our square should have moved 100 logical pixels on this iteration
+    squarePos = squarePos.translate(square_speed * squareDirection * dt, 0);
+
+    // This simple condition verifies if the square is going right, and has reached the end of the screen
+    // if so, we just invert our direction
+    //
+    // Note here that we have used the variable size, which is a variable provided
+    // by the Game class which contains the size in logical pixels that the game is currently using
+    if (squareDirection == 1 && squarePos.right > size.x) {
+      squareDirection = -1;
+    // This does the same, but now checking the left direction
+    } else if (squareDirection == -1 && squarePos.left < 0) {
+      squareDirection = 1;
+    }
+  }
+```
+
```

Comment:
I would add a penultimate paragraph saying, now we run again and get the rectangle bouncing off as we wanted!

Reply:
Done